x-pack/plugins/ml/public/jobs/new_job_new/common/job_creator/job_creator.ts

```diff
@@ -0,0 +1,226 @@
+    return this._influencers;
+  }
+
+  public set jobId(jobId: JobId) {
+    this._job_config.job_id = jobId;
+    this._datafeed_config.job_id = jobId;
+    this._datafeed_config.datafeed_id = `datafeed-${jobId}`;
+
+    if (this._useDedicatedIndex) {
+      this._job_config.results_index_name = jobId;
+    }
+  }
+
+  public get jobId(): JobId {
+    return this._job_config.job_id;
+  }
+
+  public get datafeedId(): DatafeedId {
+    return this._datafeed_config.datafeed_id;
+  }
+
+  public set description(description: string) {
+    this._job_config.description = description;
+  }
+
+  public get description(): string {
+    return this._job_config.description;
+  }
+
+  public addGroup(group: string) {
+    if (this._job_config.groups.includes(group) === false) {
+      this._job_config.groups.push(group);
+    }
+  }
+
+  public get groups(): string[] {
+    return this._job_config.groups;
+  }
+
+  public set modelPlot(enable: boolean) {
+    if (enable) {
+      this._job_config.model_plot_config = {
+        enabled: true,
+      };
+    } else {
+      delete this._job_config.model_plot_config;
+    }
+  }
+
+  public get modelPlot() {
+    return (
+      this._job_config.model_plot_config !== undefined &&
+      this._job_config.model_plot_config.enabled === true
+    );
+  }
+
+  public set useDedicatedIndex(enable: boolean) {
+    this._useDedicatedIndex = enable;
+    if (enable) {
+      this._job_config.results_index_name = this._job_config.job_id;
+    } else {
+      delete this._job_config.results_index_name;
+    }
+  }
+
+  public get useDedicatedIndex(): boolean {
+    return this._useDedicatedIndex;
+  }
+
+  public set modelMemoryLimit(mml: number | string | null) {
+    if (mml !== null) {
+      this._job_config.analysis_limits = {
+        model_memory_limit: mml,
+      };
+    } else {
+      delete this._job_config.analysis_limits;
+    }
+  }
+
+  public get modelMemoryLimit(): number | string | null {
+    if (this._job_config.analysis_limits) {
+      return this._job_config.analysis_limits.model_memory_limit;
+    } else {
+      return null;
+    }
+  }
+
+  public setTimeRange(start: number, end: number) {
+    this._start = start;
+    this._end = end;
+  }
+
+  public get start(): number {
+    return this._start;
+  }
+
+  public get end(): number {
+    return this._end;
+  }
+
+  public get query(): object {
+    return this._datafeed_config.query;
+  }
+
+  public set query(query: object) {
+    this._datafeed_config.query = query;
+  }
+
+  public get subscribers(): ProgressSubscriber[] {
+    return this._subscribers;
+  }
+
+  public async createAndStartJob() {
+    try {
+      await this.createJob();
+      await this.createDatafeed();
+      await this.startDatafeed();
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  public async createJob(): Promise<object> {
+    try {
+      return await mlJobService.saveNewJob(this._job_config);
+    } catch (error) {
+      throw error;
```

Comment:
Any reason we're catching and then throwing this error? Maybe we could add something to it to make it easier to identify where it's coming from before it's thrown?

Reply:
Yeah i was originally logging the error and then throwing, but I had to remove the log statement.
I had noted the need to improve the error handling in the PR description
- Improve job creation and starting error handling
I think i'd prefer to keep these try/catches in as a reminder that something else has to happen here. I'm thinking each function should return a success/fail and not throw at all.